runtime/src/accounts_db.rs

```diff
@@ -1947,31 +1947,43 @@ impl AccountsDb {
 
         let mut index_read_elapsed = Measure::start("index_read_elapsed");
         let mut alive_total = 0;
-        let alive_accounts: Vec<_> = {
-            stored_accounts
-                .iter()
-                .filter(|(pubkey, stored_account)| {
-                    if let Some(locked_entry) = self.accounts_index.get_account_read_entry(pubkey) {
-                        let is_alive = locked_entry.slot_list().iter().any(|(_slot, i)| {
-                            i.store_id == stored_account.store_id
-                                && i.offset == stored_account.account.offset
-                        });
-                        if !is_alive {
-                            // This pubkey was found in the storage, but no longer exists in the index.
-                            // It would have had a ref to the storage from the initial store, but it will
-                            // not exist in the re-written slot. Unref it to keep the index consistent with
-                            // rewriting the storage entries.
-                            locked_entry.unref()
-                        } else {
-                            alive_total += stored_account.account_size as u64;
-                        }
-                        is_alive
+        let mut read_lock = None;
+        if is_startup {
```

Comment:
Maybe a comment on why this is more acceptable for startup, i.e., no replay/RPC thread to contend with?